operator/src/main/resources/scripts/introspectDomain.py

```diff
@@ -760,23 +745,99 @@ def addNonClusteredServers(self):
     self.undent()
 
-  def addNetworkAccessPoints(self, server):
+  def addNetworkAccessPoints(self, server, is_server_template=False):
+    """
+    Add network access points for server or server template
+    :param server:  server or server template
+    """
     naps = server.getNetworkAccessPoints()
-    if len(naps) == 0:
-      return
-    self.writeln("  networkAccessPoints:")
-    self.indent()
-    for nap in naps:
-      self.addNetworkAccessPoint(nap)
-    self.undent()
+    added_nap = False
+    if len(naps) != 0:
+      added_nap = True
+      self.writeln("  networkAccessPoints:")
+      self.indent()
+      for nap in naps:
+        self.addNetworkAccessPoint(nap)
+
+    self.addIstioNetworkAccessPoints(server, is_server_template, added_nap)
+    if len(naps) != 0:
+      self.undent()
 
   def addNetworkAccessPoint(self, nap):
-    name=self.name(nap)
+
+    # Change the name to follow the istio port naming convention
+    istio_enabled = self.env.getEnvOrDef("ISTIO_ENABLED", "false")
+
+    if istio_enabled == 'true':
+      http_protocol = [ 'http' ]
+      https_protocol = ['https','admin']
+      tcp_protocol = [ 't3', 'snmp', 'ldap', 'cluster-broadcast', 'iiop']
+      tls_protocol = [ 't3s', 'iiops', 'cluster-broadcast-secure']
+      if nap.getProtocol() in http_protocol:
+        name = 'http-' + nap.getName().replace(' ', '_')
+      elif nap.getProtocol() in https_protocol:
+        name = 'https-' + nap.getName().replace(' ', '_')
+      elif nap.getProtocol() in tcp_protocol:
+        name = 'tcp-' + nap.getName().replace(' ', '_')
+      elif nap.getProtocol() in tls_protocol:
+        name = 'tls-' + nap.getName().replace(' ', '_')
+      else:
+        name = 'tcp-' + nap.getName().replace(' ', '_')
+    else:
+      name=self.name(nap)
     self.writeln("  - name: " + name)
     self.writeln("    protocol: " + self.quote(nap.getProtocol()))
     self.writeln("    listenPort: " + str(nap.getListenPort()))
     self.writeln("    publicPort: " + str(nap.getPublicPort()))
 
+  def addIstioNetworkAccessPoints(self, server, is_server_template, added_nap):
+    '''
+    Write the container ports information for operator to create the container ports
+    :param server:   server or template mbean
+    :param is_server_template:  true if it is from ServerTemplate
+    :param added_nap:  true if there are existing nap section in the output
+    '''
+    istio_enabled = self.env.getEnvOrDef("ISTIO_ENABLED", "false")
+    if istio_enabled == 'false':
+      return
+
+    if not added_nap:
+      self.writeln("  networkAccessPoints:")
+      self.indent()
+
+    # istio probe is not exposed
+    # istio_readiness_port = self.env.getEnvOrDef("ISTIO_READINESS_PORT", None)
+    # # self.addIstioNetworkAccessPoint("http-probe", "http", istio_readiness_port, 0)
+
+    self.addIstioNetworkAccessPoint("tcp-ldap", "ldap", server.getListenPort(), 0)
+    self.addIstioNetworkAccessPoint("tcp-default", "t3", server.getListenPort(), 0)
```

Comment:
This doesn't match the doc at the top...

|tcp-t3|server listening port|t3|Y|
|http-default|server listening port|http|Y|
|tcp-snmp|server listening port|snmp|Y|
|tcp-cbt|server listening port|CLUSTER-BROADCAST|N|
|tcp-iiop|server listening port|http|N|